pkg/client/strategies_aws.go

```diff
@@ -0,0 +1,266 @@
+	return string(marshalledStrategy), nil
+}
+
+// reconciles Redis strategy
+// unmarshalls raw strategy to create db instance input
+// checks found values from current config map vs expected values
+// marshalls create db instance input back to raw strategy
+func reconcileRedisStrategy(config, tier, backupTimeStart, maintenanceTimeStart string) (string, error) {
+	// unmarshall config data to type of cro aws strategy config
+	var rawStrategy map[string]*croAWS.StrategyConfig
+	if err := json.Unmarshal([]byte(config), &rawStrategy); err != nil {
+		return "", errorUtil.Wrapf(err, "failed to unmarshal strategy mapping for postgres resource")
+	}
+
+	// unmarshall the create strategy from the raw strategy tier to type create db instance input
+	elasticacheCreateConfig := &elasticache.CreateReplicationGroupInput{}
+	if err := json.Unmarshal(rawStrategy[tier].CreateStrategy, elasticacheCreateConfig); err != nil {
+		return "", errorUtil.Wrapf(err, "failed to unmarshal aws rds cluster config")
+	}
+
+	// check create db instance input against expected values
+	// snapshot window and maintenance window are expected values via RHMIconfig
+	// todo we may want to handle more strategy input, we should add functionality to check every value
+	if elasticacheCreateConfig.SnapshotWindow == nil || *elasticacheCreateConfig.SnapshotWindow != backupTimeStart {
+		elasticacheCreateConfig.SnapshotWindow = aws.String(backupTimeStart)
+	}
+	if elasticacheCreateConfig.PreferredMaintenanceWindow == nil || *elasticacheCreateConfig.PreferredMaintenanceWindow != maintenanceTimeStart {
+		elasticacheCreateConfig.PreferredMaintenanceWindow = aws.String(maintenanceTimeStart)
+	}
+
+	// marshall the create db instance input struct to json
+	rdsMarshalledConfig, err := json.Marshal(elasticacheCreateConfig)
+	if err != nil {
+		return "", errorUtil.Wrapf(err, "failed to marshal aws rds cluster config")
+	}
+
+	// set the createstrategy on our expected tier to the marshalled create db instance input struct
+	rawStrategy[tier].CreateStrategy = rdsMarshalledConfig
+
+	// marshall the entire strategy back to json
+	marshalledStrategy, err := json.Marshal(rawStrategy)
+	if err != nil {
+		return "", errorUtil.Wrapf(err, "failed to marshal aws strategy")
+	}
+
+	// return json in string format
+	return string(marshalledStrategy), nil
+}
+
+// builder function to provide default data for aws strategy config map
+func buildDefaultAWSConfigMap() map[string]string {
+	return map[string]string{
+		blobstorageStratKey: buildDefaultAWSStratValue(),
+		redisStratKey:       buildDefaultAWSStratValue(),
+		postgresStratKey:    buildDefaultAWSStratValue(),
+	}
+}
+
+// builder function to provide default value used per tier in aws strategy config map
+func buildDefaultAWSStratValue() string {
+	return "{\"development\": { \"region\": \"\", \"createStrategy\": {}, \"deleteStrategy\": {} }, \"production\": { \"region\": \"\", \"createStrategy\": {}, \"deleteStrategy\": {} }}"
```

Comment:
We'll now have this duplicated, once in the provider and once here. If refactoring would take too much time, can we create a follow-up ticket and link it here?